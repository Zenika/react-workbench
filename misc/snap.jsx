/* eslint-env jest */
import React from 'react'
import renderer from 'react-test-renderer'

const component = Component => (props) => {
  return renderer.create(
    <Component {...props} />,
  )
}

export default (Component) => {
  const WrappedComponent = component(Component)

  return props => () => {
    const c = WrappedComponent(props)
    const tree = c.toJSON()
    expect(tree).toMatchSnapshot()
  }
}
