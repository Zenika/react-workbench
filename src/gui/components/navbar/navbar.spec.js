/* eslint-env jest */
import snap from 'misc/test/snap'
import Navbar from './navbar'

// mock input component into props
jest.mock('../navButton', () => 'navButton')

const snapshot = snap(Navbar)

describe('component/navbar', () => {
  it('should take custom className', snapshot({ className: 'custom' }))
  it('should take custom style', snapshot({ style: { backgroundColor: 'red' } }))
  it('should have a default behaviour', snapshot({ }))
})
