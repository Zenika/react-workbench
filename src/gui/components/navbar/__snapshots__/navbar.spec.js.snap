// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component/navbar should have a default behaviour 1`] = `
<div
  className="navbar"
  style={Object {}}
>
  Workbench
</div>
`;

exports[`component/navbar should take custom className 1`] = `
<div
  className="navbar custom"
  style={Object {}}
>
  Workbench
</div>
`;

exports[`component/navbar should take custom style 1`] = `
<div
  className="navbar"
  style={
    Object {
      "backgroundColor": "red",
    }
  }
>
  Workbench
</div>
`;
